src/planner/planner.c

```diff
@@ -225,10 +225,16 @@ planner_hcache_pop(bool release)
 
 	hcache = linitial(planner_hcaches);
 
+	planner_hcaches = list_delete_first(planner_hcaches);
+
 	if (release)
+	{
 		ts_cache_release(hcache);
-
-	planner_hcaches = list_delete_first(planner_hcaches);
+		/* If we pop a stack and discover a new hypertable cache, the basrel
+		 * cache can contain invalid entries, so we reset it. */
+		if (planner_hcaches && hcache != linitial(planner_hcaches))
```

Comment:
Nit: I would compare `planner_hcaches` with `NIL` here.

`linitial` calls `list_nth_cell` which performs an `Assert(list != NIL);`.

```suggestion
		if (planner_hcaches != NIL && hcache != linitial(planner_hcaches))
```